### ISSUE TITLE
npm bugs fsevents

### ISSUE THREAD

**Issue by xuxiaotun** [Microsoft/vscode#7171]
C:\Users\EWWEE\Documents\GitHub\Jemma [master ≡ +1 ~0 -0 !]> cd vscode
C:\Users\EWWEE\Documents\GitHub\Jemma\vscode [master ≡]> scripts\npm install
npm WARN deprecated graceful-fs@3.0.8: graceful-fs v3.0.0 and before will fail o
n node releases >= v7.0. Please update to graceful-fs@^4.0.0 as soon as possible
. Use 'npm ls graceful-fs' to find it in the tree.
npm WARN deprecated lodash@1.0.2: lodash@<3.0.0 is no longer maintained. Upgrade
 to lodash@^4.0.0.
npm WARN deprecated graceful-fs@1.2.3: graceful-fs v3.0.0 and before will fail o
n node releases >= v7.0. Please update to graceful-fs@^4.0.0 as soon as possible
. Use 'npm ls graceful-fs' to find it in the tree.
npm WARN deprecated jade@0.26.3: Jade has been renamed to pug, please install th
e latest version of pug instead of jade

> fsevents@0.3.8 install C:\Users\EWWEE\Documents\GitHub\Jemma\vscode\node_modul
es\fsevents
> node-gyp rebuild

C:\Users\EWWEE\Documents\GitHub\Jemma\vscode\node_modules\fsevents>if not define
d npm_config_node_gyp (node "C:\Program Files\nodejs\node_modules\npm\bin\node-g
yp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild )  else (node "" re
build )
gyp WARN install got an error, rolling back install
gyp ERR! configure error
gyp ERR! stack Error: read ECONNRESET
gyp ERR! stack     at exports._errnoException (util.js:1007:11)
gyp ERR! stack     at TCP.onread (net.js:563:26)
gyp ERR! System Windows_NT 6.1.7601
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodej
s\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd C:\Users\EWWEE\Documents\GitHub\Jemma\vscode\node_modules\fsevents
gyp ERR! node -v v6.2.1
gyp ERR! node-gyp -v v3.3.1
gyp ERR! not ok
npm ERR! Windows_NT 6.1.7601
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\
node_modules\\npm\\bin\\npm-cli.js" "install"
npm ERR! node v6.2.1
npm ERR! npm  v3.9.3
npm ERR! code ELIFECYCLE

npm ERR! fsevents@0.3.8 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the fsevents@0.3.8 install script 'node-gyp rebuild'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the fsevents package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs fsevents
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls fsevents
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     C:\Users\EWWEE\Documents\GitHub\Jemma\vscode\npm-debug.log
C:\Users\EWWEE\Documents\GitHub\Jemma\vscode [master ≡]>

**Comment by bpasero** [comment]
@xuxiaotun are you using node.js 5 or 6 with their latest npm v3.x?

**Comment by tht13** [comment]
The log states
```
npm ERR! node v6.2.1
npm ERR! npm v3.9.3
```
I have had this problem on windows recently, fsevents is a OSX only module and npm has been having bugs determining OS compatibility/dependency of modules. This was solved but there seems to have been a regression in 3.8/9.
I solved it by rolling back to 3.7.5
`npm install -g npm@3.7.5`
It should build fine with this

If this bug gets fixed again in 3.10/4.0 you will have to remember to remove the version lock on the npm module as it will not automatically update.
